iceoryx_utils/include/iceoryx_utils/cxx/vector.hpp

```diff
@@ -155,11 +162,18 @@ class vector
     /// @brief removes the last element of the vector; calling pop_back on an empty container does nothing
     void pop_back();
 
-    /// @brief removes an element at the given position. if this element is in
+    /// @brief removes an element at the given iterator position. if this element is in
     ///         the middle of the vector every element is moved one place to the
     ///         left to ensure that the elements are stored contiguously
+    /// @return iterator to the next element, returns nullptr if position is out-of-range
     iterator erase(iterator position);
 
+    /// @brief removes an element at the given vector index. if this element is in
+    ///         the middle of the vector every element is moved one place to the
+    ///         left to ensure that the elements are stored contiguously
+    /// @return iterator to the next element, returns nullptr if index is out-of-range
+    iterator erase(uint64_t index);
```

Comment:
The method of std::vector is "iterator erase( const_iterator pos )". It does not have "noexcept", since an assignment operation of the underlying data type could potentially throw an exception. I propose to have "noexcept" either on both erase methods or on none. What do you think?

Reply:
We essentially write on every method `noexcept`. You are right, it is possible that the underlying assignment operator of a certain type throws an exception but in general we do not support exceptions of any kind. And with the noexcept we state it clearly to the public - so please add it to the method.
But maybe we should also enforce it even more with a `static_assert` in the vector to only allow types which are noexcept copy and movable. @elBoberido what do you think?

Reply:
Good question. The big problem I see here is a third party class where you do not have the possibility to adapt it, but you know it cannot throw an exception. It's just the developer who forgot to add the keyword. I think it is sufficient to have it in the documentation.

Reply:
Done (documented and without noexcept).